src/dmd/access.d

```diff
@@ -144,13 +144,25 @@ extern (C++) bool checkAccess(AggregateDeclaration ad, Loc loc, Scope* sc, Dsymb
         printf("AggregateDeclaration::checkAccess() for %s.%s in function %s() in scope %s\n", ad.toChars(), smember.toChars(), f ? f.toChars() : null, cdscope ? cdscope.toChars() : null);
     }
     Dsymbol smemberparent = smember.toParent();
```

Comment:
If it weren't for the deprecation, the single modification for this patch would have been transforming this line into:

`Dsymbol smemberparent = smember.toParent2()`

Explanation:
For template declarations, the parent is the enclosing template declaration; we are not interested in that in this situation, but if the template declaration itself is nested by a struct or not.